test/internal/_validator_test.dart

```diff
@@ -76,6 +77,109 @@ void main() {
           isTrue,
         );
       });
+
+      // Portugal supports all phone types
+      test('PT', () {
+        final isoCode = IsoCode.PT;
+        final examples = metadataExamplesByIsoCode[isoCode]!;
+        expect(
+          Validator.validateWithLength(
+            isoCode,
+            examples.mobile,
+          ),
+          isTrue,
+        );
+
+        expect(
+          Validator.validateWithLength(
+            isoCode,
+            examples.mobile,
+            PhoneNumberType.mobile,
+          ),
+          isTrue,
+        );
+
+        expect(
+          Validator.validateWithLength(
+            isoCode,
+            examples.fixedLine,
+            PhoneNumberType.fixedLine,
+          ),
+          isTrue,
+        );
+
+        expect(
+          Validator.validateWithLength(
+            isoCode,
+            examples.voip,
+            PhoneNumberType.voip,
+          ),
+          isTrue,
+        );
+
+        expect(
+          Validator.validateWithLength(
+            isoCode,
+            examples.tollFree,
+            PhoneNumberType.tollFree,
+          ),
+          isTrue,
+        );
+
+        expect(
+          Validator.validateWithLength(
+            isoCode,
+            examples.premiumRate,
+            PhoneNumberType.premiumRate,
+          ),
+          isTrue,
+        );
+
+        expect(
+          Validator.validateWithLength(
+            isoCode,
+            examples.sharedCost,
+            PhoneNumberType.sharedCost,
+          ),
+          isTrue,
+        );
+
+        expect(
+          Validator.validateWithLength(
+            isoCode,
+            examples.personalNumber,
+            PhoneNumberType.personalNumber,
+          ),
+          isTrue,
+        );
+
+        expect(
+          Validator.validateWithLength(
+            isoCode,
+            examples.uan,
+            PhoneNumberType.uan,
+          ),
+          isTrue,
+        );
+
+        expect(
+          Validator.validateWithLength(
+            isoCode,
+            examples.pager,
+            PhoneNumberType.pager,
+          ),
+          isTrue,
+        );
+
+        expect(
```

Comment:
To cover more surface it would be nice to add a negative test somewhere too. Something like:

```dart
        expect(
          Validator.validateWithLength(
            isoCode,
            examples.pager,     // assuming the pager example is not a valid mobile 
            PhoneNumberType.mobile,
          ),
          isFalse,
        );
```

Reply:
Alright, I'll add some